lib/pbench/server/auth/auth.py

```diff
@@ -1,25 +1,95 @@
+from dataclasses import dataclass
 import datetime
 from http import HTTPStatus
 import os
-from typing import Optional
+from typing import Optional, Union
 
 from flask import abort, request
 from flask_httpauth import HTTPTokenAuth
 import jwt
 
+from pbench.server import PbenchServerConfig
 from pbench.server.auth import OpenIDClient, OpenIDClientError
 from pbench.server.database.models.active_tokens import ActiveTokens
 from pbench.server.database.models.users import User
 
 
+@dataclass
+class InternalUser:
+    """Internal user class for storing user related fields fetched
+    from OIDC token decode.
+    """
+
+    id: str
+    username: str
+    email: str
+    first_name: Optional[str] = None
+    last_name: Optional[str] = None
+    roles: Optional[list[str]] = None
```

Comment:
Might be worth a mention in the docstring that these attributes are duck-typed from the SQL `User` object and absolutely need to match!